src/components/Search/SearchViewInit.tsx

```diff
@@ -1,16 +1,17 @@
 import { useEffect } from 'react';
 
-import { connect, useDispatch } from 'react-redux';
+import { connect } from 'react-redux';
 
+import { useAppDispatch } from '../../hooks/app';
 import { fetchRecentRevisions } from '../../thunks/searchThunk';
-import { Repository, State } from '../../types/state';
+import type { Repository, State } from '../../types/state';
 
 // component to fetch recent revisions when search view is loaded
 function SearchViewInit(props: SearchViewInitProps) {
-  const dispatch = useDispatch();
+  const dispatch = useAppDispatch();
```

Comment:
https://redux-toolkit.js.org/tutorials/typescript#define-typed-hooks

without using a typed hook, I was getting these errors:
```
TS2345: Argument of type 'AsyncThunkAction<Revision[], "mozilla-central" | "mozilla-beta" | "mozilla-release" | "try" | "fenix" | "autoland", { rejectValue: string; }>' is not assignable to parameter of type 'AnyAction'.
    13 |   const { repository } = props;
    14 |   useEffect(() => {
  > 15 |     void dispatch(fetchRecentRevisions(repository));
       |                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    16 |   }, []);
    17 |   return null;
    18 | }
```

Reply:
it's new for me to see void on a function call.
I guess this is how react handles return type now ?

Reply:
This is a standard javascript operator: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/void

Basically this "voids" the value on the right. It's not common to use it directly (at least _I_ don't use it often if any :-) ), however you can see this in minified scripts, because minifiers usually replace `undefined` by `void 0` because it's shorter.